packages/origin-ui-core/src/components/bundles/Certificates.tsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+import {
+    Paper,
+    List,
+    ListSubheader,
+    ListItem,
+    ListItemIcon,
+    Checkbox,
+    ListItemText,
+    Grid,
+    Typography,
+    ListItemAvatar,
+    Avatar
+} from '@material-ui/core';
+import { useSelector } from 'react-redux';
+import moment from 'moment';
+import { getCertificates, ICertificateViewItem } from '../../features/certificates';
+import {
+    getProducingDevices,
+    getEnvironment,
+    deviceById,
+    PowerFormatter,
+    energyImageByType
+} from '../..';
+
+interface IOwnProps {
+    selected: ICertificateViewItem[];
```

Comment:
Alternatively, to avoid all array finding operations, you could create a type that extends `ICertificateViewItem` with `selected: boolean` property. This would allow you to flip `selected` flag instead of finding and replacing in the `selected` array.